buildtools/AssemblyCheck/AssemblyCheck.fsproj

```diff
@@ -8,7 +8,8 @@
 
   <PropertyGroup>
     <PublishReadyToRun>true</PublishReadyToRun>
-    <UseCurrentRuntimeIdentifier>true</UseCurrentRuntimeIdentifier>
+    <UseCurrentRuntimeIdentifier Condition="'$(DotNetBuildFromSource)' != 'true'">true</UseCurrentRuntimeIdentifier>
+    <RuntimeIdentifier Condition="'$(DotNetBuildFromSource)' == 'true'">$(TargetRid)</RuntimeIdentifier>
```

Comment:
I think these two lines can be simplified to:

```xml
<RuntimeIdentifier>$(NETCoreSdkRuntimeIdentifier)</RuntimeIdentifier>
```

Reply:
Is the NETCoreSdkRuntimeIdentifier correct when doing a stage1 product build? Isn't that linux-x64 then?

Reply:
> Is the NETCoreSdkRuntimeIdentifier correct when doing a stage1 product build and using the MSFT SDK? Isn't that linux-x64 then?

I hadn't fully considered this.

We should use `TargetRid` when we want to use what was part of the current build and meant for the target platform, and `NETCoreSdkRuntimeIdentifier` when we want what came with the build SDK and compatible with the build platform.

Reply:
TargetRid caused a build failure before:

>   /vmr/.dotnet/sdk/9.0.100-preview.5.24258.1/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Publish.targets(400,5): error NETSDK1095: Optimizing assemblies for performance is not supported for the selected target platform or architecture. Please verify you are using a supported runtime identifier, or set the PublishReadyToRun property to false. [/vmr/src/fsharp/buildtools/fslex/fslex.fsproj] [/vmr/artifacts/source-built-sdks/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj]


That's probably because the MSFT crossgen2 is built portable for linux-x64 and doesn't understand the non-portable ubuntu22.04-x64 RID. I changed `NETCoreSdkRuntimeIdentifier` everywhere which works and is correct for the msft build and the stage 2 source-build. We can revisit this if needed.